src/pkg/logs/logrus.go

```diff
@@ -17,3 +23,44 @@ func IsLogrusError(message string) bool {
 		return true // show by default (likely Dockerfile errors)
 	}
 }
+
+type TermLogFormatter struct {
+	Term *term.Term
+}
+
+func (f TermLogFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	getMessage := func() string {
+		var buf strings.Builder
+		buf.WriteString(entry.Message)
+		for k, v := range entry.Data {
+			buf.WriteString(" ")
+			buf.WriteString(k)
+			buf.WriteString("=")
+			buf.WriteString(v.(string))
+		}
+		return buf.String()
+	}
+
+	switch entry.Level {
+	case logrus.PanicLevel, logrus.FatalLevel:
+		f.Term.Fatal(getMessage())
+	case logrus.ErrorLevel:
+		f.Term.Error(getMessage())
+	case logrus.WarnLevel:
+		f.Term.Warn(getMessage())
+	case logrus.InfoLevel:
+		f.Term.Info(getMessage())
+	case logrus.DebugLevel, logrus.TraceLevel:
+		if f.Term.DoDebug() {
```

Comment:
no need for this `if`, `Debug` does that. Sure we save on `getMessage` but feels like premature optimization

Reply:
Agreed